### ISSUE TITLE
GPU memory for training

### ISSUE THREAD

**Issue by ZeroVector1201** [thu-nics/C2C#3]
Could you please advise on how to calculate the GPU memory requirements for training models of varying sizes using the C2C method? Any insights would be greatly appreciated.

**Comment by minzh23** [comment]
Hi, thanks for the question! 👋

In our C2C experiments with the **Qwen3-0.6B + Qwen2.5-0.5B-Instruct** setup, we used the following configuration:

```json
"gradient_accumulation_steps": 8,
"per_device_train_batch_size": 4,
"max_length": 2048
```

The training was conducted on **NVIDIA A800-SXM4-80GB** GPUs.

Here are some practical notes on GPU memory requirements and scaling across different model sizes:

- **Model size vs. batch size**  
  Larger models naturally consume more GPU memory. When scaling up, you can proportionally reduce `per_device_train_batch_size`. Conversely, smaller models allow larger per-device batches.

- **Gradient accumulation**  
  Since C2C supports `gradient_accumulation_steps`, you can increase this value to simulate a larger total batch size without increasing GPU memory usage. This is the preferred approach when VRAM is limited.

- **Sequence length (`max_length`)**  
   It’s a good practice to tune `max_length` based on your dataset and filter out extremely long samples to prevent memory spikes or OOM errors.

Overall, GPU memory usage in C2C training scales with **model size**, **batch size**, and **sequence length**. The framework supports flexible adjustment of these parameters to fit different hardware setups.

**Comment by dant2021** [comment]
I've trained on 24GB 4090s with 
```
"gradient_accumulation_steps": 8,
"per_device_train_batch_size": 4,
"max_length": 1024
